### PR TITLE
atom support

### PR DESCRIPTION
**Pull request by mbenatti**
please review, I can be missing something

### CODE REVIEW

File: test/exnumerable_test.exs [hunk 1]
```diff
@@ -2,30 +2,57 @@ defmodule ExnumeratorTest do
   use ExUnit.Case
 
   @values ["sent", "read", "received", "delivered"]
+  @values_as_atoms [:sent, :read, :received, :delivered]
```
Comment (KamilLelonek) [inline test/exnumerable_test.exs]: I would recommend to use here:

```elixir
@values_strings ["sent", "read", "received", "delivered"]
@values_atoms   [:sent, :read, :received, :delivered]
```

File: test/exnumerable_test.exs [hunk 2]
```diff
@@ -2,30 +2,57 @@ defmodule ExnumeratorTest do
...
 
   defmodule Message do
     use Exnumerator,
       values: ["sent", "read", "received", "delivered"]
   end
 
+  defmodule MessageAsAtoms do
```
Comment (KamilLelonek) [inline test/exnumerable_test.exs]: I suggest renaming the other module correspondingly to keep the consistency.

File: lib/exnumerable.ex [hunk 3]
```diff
@@ -7,10 +7,18 @@ defmodule Exnumerator do
       def values, do: unquote(values)
       def sample, do: unquote(values) |> Enum.random
 
-      for value <- values do
-        def cast(unquote(value)), do: {:ok, unquote(value)}
-        def load(unquote(value)), do: {:ok, unquote(value)}
-        def dump(unquote(value)), do: {:ok, unquote(value)}
+      if Enum.all?(values, &(is_atom(&1))) do
```
Comment (KamilLelonek) [inline lib/exnumerable.ex:10]: What happens when types are mixed?

**Review by KamilLelonek — CHANGES_REQUESTED** [review]
What do you think about documenting these changes?

File: lib/exnumerable.ex (hunk 3 above)
Comment (mbenatti) [reply lib/exnumerable.ex:10]: will get :error, nothing change for this, the same behavior

File: test/exnumerable_test.exs (hunk 2 above)
Comment (mbenatti) [reply test/exnumerable_test.exs]: done

File: test/exnumerable_test.exs (hunk 1 above)
Comment (mbenatti) [reply test/exnumerable_test.exs]: done

**Review by KamilLelonek — COMMENTED** [review]
Can you document your changes in the README?

**Review by KamilLelonek — APPROVED** [review]

**Comment by mbenatti** [comment]
@KamilLelonek before you generate another version in hex I have another improvement...

**Pull request merged**